src/Sparrow/Json/UnmanagedWriteBuffer.cs

```diff
@@ -272,20 +286,24 @@ public void Clear()
 
         public void Dispose()
         {
-            while (_current != null)
-            {
+            while (_current != null && 
```

Comment:
There is no need here, we are already running over this and setting it to null on the root object.